external-import/socradar/src/lib/radar.py

```diff
@@ -0,0 +1,354 @@
+        }
+
+    def _get_indicator_type(self, feed_type):
+        """Map feed types to STIX indicator types"""
+        type_mapping = {
+            "url": ["url-watchlist"],
+            "domain": ["domain-watchlist"],
+            "ip": ["ip-watchlist"],
+            "hash": ["file-hash-watchlist"],
+        }
+        return type_mapping.get(feed_type, ["malicious-activity"])
+
+    def _get_or_create_identity(self, maintainer_name):
+        """Get existing identity or create new one for maintainer"""
+        try:
+            if maintainer_name in self.identity_mapping:
+                return self.identity_mapping[maintainer_name]
+
+            current_time = datetime.utcnow()
+
+            # Create new identity for maintainer
+            identity = Identity(
+                id=f"identity--{str(uuid.uuid4())}",
+                name=maintainer_name,
+                identity_class="organization",
+                description=f"Feed Provider: {maintainer_name}",
+                sectors=["technology"],
+                created=current_time,
+                modified=current_time,
+            )
+
+            # Store in mapping for reuse
+            self.identity_mapping[maintainer_name] = identity
+            self.helper.log_info(
+                f"Created new identity for maintainer: {maintainer_name}"
+            )
+            return identity
+
+        except Exception as e:
+            self.helper.log_error(
+                f"Error creating identity for {maintainer_name}: {str(e)}"
+            )
+            return None
+
+    def _process_feed_item(self, item, work_id):
+        try:
+            value = item["feed"]
+            feed_type = item.get("feed_type", "").lower()
+            maintainer = item.get("maintainer_name", "Unknown")
+            first_seen = datetime.strptime(item["first_seen_date"], "%Y-%m-%d %H:%M:%S")
+            last_seen = datetime.strptime(item["latest_seen_date"], "%Y-%m-%d %H:%M:%S")
+
+            if last_seen <= first_seen:
+                last_seen = first_seen + timedelta(days=365)
+
+            # Get or create identity for this maintainer
+            maintainer_identity = self._get_or_create_identity(maintainer)
+            if not maintainer_identity:
+                self.helper.log_error(
+                    f"Could not create identity for maintainer: {maintainer}"
+                )
+                return
+
+            pattern = self._create_stix_pattern(value, feed_type)
+            if not pattern:
+                self.helper.log_error(
+                    f"Could not create pattern for: {value} ({feed_type})"
+                )
+                return
+
+            # Create kill chain phase
+            kill_chain_phase = KillChainPhase(
+                kill_chain_name="lockheed-martin-cyber-kill-chain",
+                phase_name="reconnaissance",
+            )
+
+            indicator = Indicator(
+                id=f"indicator--{str(uuid.uuid4())}",
```

Comment:
This would create stochastic id and will prevent de-duplication performed by the platform.

You should rather use pycti.Indicator.generate_id method.